components/data_comps/docn/src/docn_comp_mod.F90

```diff
@@ -768,6 +786,72 @@ subroutine docn_comp_run(EClock, x2o, o2x, &
           enddo
        endif   ! firstcall
 
+    ! Relaxed Slab Ocean based on Zarzycki(2016)
+    ! Zarzycki, C. M., 2016: Tropical Cyclone Intensity Errors Associated with Lack of Two-Way Ocean Coupling in High-Resolution Global Simulations. J. Climate, 29, 8589–8610.
+    ! https://journals.ametsoc.org/view/journals/clim/29/23/jcli-d-16-0273.1.xml
+    case('RSO')
+      lsize = mct_avect_lsize(o2x)
+      do n = 1,SDOCN%nstreams
+        call shr_dmodel_translateAV(SDOCN%avs(n),avstrm,avifld,avofld,rearr)
+      enddo
+      if (firstcall) then
+        do n = 1,lsize
+          if (.not. read_restart) then
+            somtp(n) = o2x%rAttr(kt,n) + TkFrz
+          endif
+          o2x%rAttr(kt,n) = somtp(n)
+          o2x%rAttr(kq,n) = 0.0_R8
+        enddo
+      else   ! firstcall
+        tfreeze = shr_frz_freezetemp(o2x%rAttr(ks,:)) + TkFrz
+        do n = 1,lsize
+          if (imask(n) /= 0) then
+            !*******************************************************************
+            if (RSO_fixed_MLD>=0) then
+              hn = RSO_fixed_MLD
+            else
+              hn = avstrm%rAttr(kh,n)
+            endif
+            ! Get "background" temperature for relaxation
+            RSO_bckgrd_sst = avstrm%rAttr(kRSO_bckgrd_sst,n) + TkFrz
+            u10 = SQRT(x2o%rAttr(k10uu,n))
+            !*******************************************************************
+            ! RSO parameter values
+            RSO_slab_option = 0                  ! Option for setting RSO_X_cool
+            RSO_R_cool      = 11.75_r8/86400._r8 ! base cooling rate [K/s]
+            RSO_Tdeep       = 271.00             ! deep water temperature [K]
+            RSO_dT_o        = 27.0               ! scaling temperature gradient
+            RSO_h_o         = 30.0               ! scaling mixed layer depth
```

Comment:
Thanks for fixing some of these. If you do not want to add `_r8`, the numeric literals should be assigned to variables as integers to maintain precision. Assigning variables with numbers with decimals (e.g., 30.0) will reduce their precision to single precision.
```suggestion
            RSO_Tdeep       = 271                ! deep water temperature [K]
            RSO_dT_o        = 27                 ! scaling temperature gradient
            RSO_h_o         = 30                 ! scaling mixed layer depth
```

Being explicit and adding `_r8` (e.g., `30.0_r8`) is still a better approach, as it will make it clear to the reader that these are real numbers. Declaring them as parameters is also preferable if these variables will stay constant during time-stepping.

Reply:
ok, I like suggestions.

Reply:
Done.